driver-mapping/src/main/java/com/datastax/driver/mapping/config/DefaultHierarchyScanStrategy.java

```diff
@@ -0,0 +1,103 @@
+/*
+ *      Copyright (C) 2012-2015 DataStax Inc.
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License");
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ */
+package com.datastax.driver.mapping.config;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+/**
+ * The default {@link HierarchyScanStrategy}.
+ * <p/>
+ * This strategy assumes that there exists a common ancestor
+ * for all mapped classes in the application, and allows all its
+ * descendants (optionally including itself) to be scanned for annotations.
+ * <p/>
+ * By default, the common ancestor is {@link Object}, which implies
+ * that every ancestor of a mapped class, except {@code Object} itself,
+ * will be scanned for annotations.
+ */
+public class DefaultHierarchyScanStrategy implements HierarchyScanStrategy {
+
+    /**
+     * Returns a new {@link DefaultHierarchyScanStrategy.Builder} instance.
+     *
+     * @return a new {@link DefaultHierarchyScanStrategy.Builder} instance.
+     */
+    public static DefaultHierarchyScanStrategy.Builder builder() {
+        return new DefaultHierarchyScanStrategy.Builder();
+    }
+
+    /**
+     * Builder for {@link DefaultHierarchyScanStrategy} instances.
+     */
+    public static class Builder {
```

Comment:
When I added a builder here it was mainly for consistency with other configuration classes. Since I'm now advocating for removing the builders elsewhere, this one could go too.